### ISSUE TITLE
Segmentation fault when training index with 1.7.2 on macOs 12.2.1 x86

### ISSUE THREAD

**Issue by jobergum** [facebookresearch/faiss#2410]
# Summary

I'm experiencing segmentation faults when training the index 

# Platform

OS: macOS 12.2.1 x86 

Faiss version: 1.7.2

```
pip3 install --upgrade faiss_cpu
Requirement already satisfied: faiss_cpu in /usr/local/anaconda3/lib/python3.8/site-packages (1.7.2)
```

Running on:
- CPU (x86) 

Interface: 
- Python 3.8.5 

# Reproduction instructions

Download sample vectors (1.3GiB) 
```
wget https://mystic.the-eye.eu/public/AI/cah/laion5b/embeddings/laion2B-en/img_emb/img_emb_0000.npy
```

Run the following snippet, I've tried multiple index factory variants. Everything except "Flat" fails with a segmentation fault. The image embedding vectors are L2 normalized, so the inner product equals cosine similarity. 

```
import faiss
import numpy as np

vectors = np.load('img_emb_0000.npy').astype(np.float32)
index = faiss.index_factory(vectors.shape[1], "IVF4096,PQ32x8", faiss.METRIC_INNER_PRODUCT)
index.train(vectors)
```

I've also tried with fewer vectors to see if this was related to going out of memory:
```
import faiss
import numpy as np

vectors = np.load('img_emb_0000.npy').astype(np.float32)
index = faiss.index_factory(vectors.shape[1], "IVF4096,PQ32x8", faiss.METRIC_INNER_PRODUCT)
index.train(vectors[0:10000])
```

Which also crashes:
```
python3 index.py data/img_emb_0000.npy 
WARNING clustering 10000 points to 4096 centroids: please provide at least 159744 training points
Segmentation fault: 11
```

Thank you for any pointers on this.

**Comment by Ben-Epstein** [comment]
+1 I am seeing the same thing. Failing on MacOS, working on linux

Downgrading to `faiss-cpu==1.7.0` fixes my issue (for anyone in desperate need of a fix)

**Comment by ldstefanou** [comment]
Same for me on an M1, also only seems to happen on `1.7.2`. Downgrading worked for me.

**Comment by jonathangomesselman** [comment]
+1 same for me on an Intel MacOS. Similar thing where downgrading worked then for me.

**Comment by mdouze** [comment]
My assumption is that it's related to installing via pip rather than anaconda. 
Could you check with a clean anaconda install?

**Comment by jobergum** [comment]
You are right @mdouze, thank you very much for your support. The above snippets works after installing `faiss-cpu` by conda instead of pip.

For others finding this issue, install using [anaconda faiss-cpu](https://anaconda.org/pytorch/faiss-cpu) instead.

**Comment by Ben-Epstein** [comment]
@jobergum @mdouze are you sure it's fair to close this issue? It seems a bit dangerous to keep out a version of a package that is seemingly stable but causes a segmentation fault when installed using `pip` instead of `conda`

Is this to say that `pip` isn't officially supported?

**Comment by mdouze** [comment]
pip is not supported for precisely this reason. 
We cannot disable install through pip
Besides, let's face it: often pip install happens to just work...

**Comment by doughazell** [comment]
Thanks this solved the seg fault problem on MacOS Catalina when running http://docs.deeppavlov.ai/en/master/features/models/odqa.html (which worked fine on Google Colab and native Ubuntu)

**Comment by chn-lee-yumi** [comment]
1.13.0 still not works too...
